core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -194,11 +212,20 @@ class GroupCoordinator(val brokerId: Int,
       } else if (!group.supportsProtocols(protocolType, MemberMetadata.plainProtocolSet(protocols))) {
         responseCallback(joinError(memberId, Errors.INCONSISTENT_GROUP_PROTOCOL))
       } else if (group.isPendingMember(memberId)) {
-        // A rejoining pending member will be accepted.
-        addMemberAndRebalance(rebalanceTimeoutMs, sessionTimeoutMs, memberId, clientId, clientHost, protocolType,
-          protocols, group, responseCallback)
+        // A rejoining pending member will be accepted. Note that pending member will never be a static member.
+        assert(groupInstanceId == JoinGroupRequest.UNKNOWN_GROUP_INSTANCE_ID)
+        addMemberAndRebalance(rebalanceTimeoutMs, sessionTimeoutMs, memberId, JoinGroupRequest.UNKNOWN_GROUP_INSTANCE_ID
+          ,clientId, clientHost, protocolType, protocols, group, responseCallback)
+      } else if (isStaticMember && !group.hasStaticMember(groupInstanceId)) {
+        // The given static member is not found within the static membership,
+        // informing member to rejoin with unknown member id.
+        responseCallback(joinError(memberId, Errors.GROUP_INSTANCE_ID_NOT_FOUND))
+      } else if (isStaticMember && group.getStaticMemberId(groupInstanceId) != memberId) {
+        // the given member id doesn't match with the groupInstanceId. Should inform
+        // duplicate instance to shut down immediately.
```

Comment:
nit: `should inform` - `inform duplicate instance...`
This keeps it consistent with the tense in the other comments